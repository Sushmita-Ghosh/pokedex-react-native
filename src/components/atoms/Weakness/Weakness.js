import React, { useState, useEffect, memo } from "react";
import { StyleSheet, View, Text } from "react-native";

import { getPokemonWeakness } from "../../../api/pokemon.js";
import Box from "../Box/Box.js";
import { useSelector } from "react-redux";

function Weakness() {
  const pokemon = useSelector((state) => state.pokemon.currentPokemon);
  const [pokemonType, setPokemonType] = useState();

  // Pokemon weakness call
  const loadPokemonTypes = async (name) => {
    const res = await getPokemonWeakness(name);
    setPokemonType(res);
  };

  useEffect(() => {
    (async () => {
      await loadPokemonTypes(
        pokemon.types[0].type.name ? pokemon.types[0].type.name : "bug"
      );
    })();
  }, [pokemon.types[0].type.name]);

  if (!pokemonType) return;

  return (
    <>
      <View style={styles.group}>
        <Text style={styles.headerText}>Weak Against</Text>
        <View
          style={{
            width: "100%",
          }}
        >
          <Box
            data={[
              ...pokemonType.damage_relations.double_damage_from,
              //   ...pokemonType.damage_relations.half_damage_from,
            ]}
            isWeak
          />
        </View>
      </View>
    </>
  );
}

const styles = StyleSheet.create({
  group: {
    width: "100%",
    justifyContent: "center",
  },

  headerText: {
    fontWeight: "bold",
    fontSize: 16,
    paddingBottom: 5,
  },
});

export default memo(Weakness);
