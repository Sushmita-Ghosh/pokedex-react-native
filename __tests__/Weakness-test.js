import "react-native";
import React from "react";
import Weakness from "../src/components/pokemon/Weakness";
import renderer from "react-test-renderer";

const pokemonProp = {
  name: "Bulbasaur",
  types: [
    {
      slot: 1,
      type: { name: "grass", url: "https://pokeapi.co/api/v2/type/12/" },
    },
  ],
};

test("render app snapshot", () => {
  const snapshot = renderer.create(<Weakness pokemon={pokemonProp} />).toJSON();

  expect(snapshot).toMatchSnapshot();
});
