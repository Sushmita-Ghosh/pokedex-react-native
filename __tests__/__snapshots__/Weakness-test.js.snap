// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render app snapshot 1`] = `null`;
